### PR TITLE
lib: nrf_cloud: add nrf_cloud_coap_bytes_send

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds a method to nRF Cloud CoAP to send raw bytes. (tested on dev stage)

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]
Looks fine.  Just needs a changelog entry.

File: subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c [hunk 1]
```diff
@@ -212,6 +212,24 @@ int nrf_cloud_coap_pgps_url_get(struct nrf_cloud_rest_pgps_request const *const
...
+
+int nrf_cloud_coap_bytes_send(uint8_t *buf, size_t buf_len)
+{
+	int err = 0;
+
+	if (!nrf_cloud_coap_is_connected()) {
+		return -EACCES;
+	}
+
+	err = nrf_cloud_coap_post("/msg/d2c/raw", NULL, buf, buf_len,
```
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c:224]: can you add `#define COAP_D2C_RAW_RSC "/msg/d2c/raw"` just to be consistent with the other resource paths?

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
Approved assuming you implement Justin's change suggestion

**Comment by glarsennordic** [comment]
adding DNM so that it doesn't accidentally get merged due to approvals

**Review by plskeggs — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -522,6 +522,7 @@ Libraries for networking
 
     * Automatic selection of proprietary PSM mode when building for the SOC_NRF9161_LACA.
     * Support for bulk transfers to the :c:func:`nrf_cloud_coap_json_message_send` function.
+    * Support for raw transfers: :c:func:`nrf_cloud_coap_bytes_send`
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:525]:
```suggestion
    * Support for raw transfers to the :c:func:`nrf_cloud_coap_bytes_send` function.
```

**Review by divipillai — APPROVED** [review]

**Pull request merged**